pkgs/java_http/lib/src/java_client.dart

```diff
@@ -102,6 +156,7 @@ class JavaClient extends BaseClient {
     final statusCode = httpUrlConnection.getResponseCode();
 
     if (statusCode == -1) {
+      // TODO: Send exception instead of throw.
       throw ClientException(
           'Status code can not be discerned from the response.', requestUrl);
     }
```

Comment:
Here we are throwing a `ClientException` in the worker isolate, I can change this in a separate PR.

Reply:
:heavy_check_mark: This has been fixed in this PR!

We are now throwing the ClientException in the main isolate instead of the worker isolate :tada:.